src/sagemaker/model.py

```diff
@@ -1307,7 +1337,19 @@ def deploy(
         )
 
         if self.predictor_cls:
-            predictor = self.predictor_cls(self.endpoint_name, self.sagemaker_session)
+            if self.predictor_cls == JumpStartPredictor:
```

Comment:
remove please: the JumpStart-specific logic shouldn't leak into the base class. Essentially, users shouldn't pass a `JumpStartPredictor` to the parent class directly.